### DISCUSSION TITLE
Annotating Images with Data from IPTC / EXIF Metadata

### DISCUSSION THREAD

**Discussion by Trailboy** [ImageMagick/ImageMagick#6107] [General] [answered]
Hi again.  

Hoping that the fantastic brains trust is able to help me again.   

Link to files: https://drive.google.com/drive/folders/1jkv4U7EjxlbwusGc_vEO5hbvgn-uk37R?usp=sharing
Version of imagemagic - 7.1.0.4 

Our school uses magick commands to annotate many images at once using the Windows .bat batch file method.  

Previously, we have managed to annotate images from the filename with the help of the wonderful people outlined here:
https://github.com/ImageMagick/ImageMagick/discussions/4054

Our current code (slightly adapted from the above) is this:
```
for %%F in (*.jpg) do magick "%%F" ^
  -resize 900x900 ^
  -gravity South ^
  -fill "rgba( 255, 255, 255, 0.9 )" ^
  -draw "rectangle 0,850 850,900" ^
  -fill black ^
  -pointsize 26  ^
  -annotate +0+10 %%t ^
  -quality 100 ^
  "%%~nF_Annotated.jpg"
```
which works very well and gives the _Annotated image as it appears in the Drive link above.  

**Question: can we use the images embedded metadata to annotate images instead of the filename?** 

Having researched on the legacy forums and this, the following code changes %%t to %%[IPTC:Instructions] but this does not print the metadata as an annotation.  

```
 for %%F in (*.jpg) do magick "%%F" ^
  -resize 900x900 ^
  -gravity South ^
  -fill "rgba( 255, 255, 255, 0.5 )" ^
  -draw "rectangle 0,850 850,900" ^
  -fill black ^
  -pointsize 26  ^
  -annotate +0+10 %%[IPTC:Instructions] ^
  -quality 100 ^
  "%%~nF_Annotated.jpg"
```

![Annotation Metadata Missing](https://user-images.githubusercontent.com/64076927/221408192-14d7001e-f65d-4f04-ae75-c99a8c6275fe.JPG)

Does anybody know the correct Syntax and why ` -annotate +0+10 %%[IPTC:Instructions] ^` isn't working for us?  We can't get any metadata field to write as an annotation using a batch file, even though it is in the image.  I suspect it is the windows batch file syntax that is tripping us up in some way, but trying multiple variations has not helped so far.

If we can get the annotation from the metadata, then we can hopefully amend the code so that it may allow us to use multiple lines of annotated data (such as name and position or class) on each image on different lines, which would be very useful to us indeed.

Thanks so much for the help now and in the past.

**Comment by fmw42** [comment]
I may be overlooking it, but I do not see %[IPTC:instructions] in the list at https://imagemagick.org/script/escape.php

And the syntax is %[IPTC:dataset:record]

**Comment by snibgo** [comment]
Naming your image `instr.JPG`:
```
C:\web\im>%IMG7%magick instr.JPG -pointsize 30 -gravity Center -annotate +0+0 %[IPTC:Instructions] x.jpg
magick: unknown image property "%[IPTC:Instructions]" @ warning/property.c/InterpretImageProperties/4236.
```
So your image doesn't seem to have that metadata.

```
C:\web\im>%IMG7%magick instr.JPG -pointsize 30 -gravity Center -annotate +0+0 %[exif:Artist] x.jpg
```
That works, and writes "Kris" on the image.

(My syntax is Windows CMD. For Windows BAT, double each `%` that is within the magick command.)

**Comment by Trailboy** [comment]
Thank you fmw42 and snibgo. 

At least I can see that the syntax was not the problem. 

After further experimenting, it does appear that certain metadata fields are readable by Imagemagick and some not: i.e. certain EXIF is, but I don't think any IPTC is.  If so, that is a pity.  I have made sure that the relevant metadata field is present in the tested images using exiftool and gimp metadata view. 

For example, [Exif:Software] and [exif:datetime] indeed work fine:

![Capture01](https://user-images.githubusercontent.com/64076927/221431736-1b808b21-3576-4117-bf81-740671b1b712.JPG)

![Capture02](https://user-images.githubusercontent.com/64076927/221431753-d0ab210a-384d-470e-bb62-01c825c2b729.JPG)

but I can't get any iptc metadata to annotate, not matter how I try:

![Capture03-](https://user-images.githubusercontent.com/64076927/221431896-6676e1bc-8397-4985-8e7c-9ee46f978b93.JPG)

![Capture04](https://user-images.githubusercontent.com/64076927/221431908-2309bfc6-3e08-49f6-a88a-b52e98703d32.JPG)

![Capture05](https://user-images.githubusercontent.com/64076927/221431922-a0eb259e-5142-4a95-8e74-e689bba1f771.JPG)

**Comment by snibgo** [comment]
Please post a sample file that contains IPTC metadata. The actual file, not a screenshot of the file.

**Comment by Trailboy** [comment]
Hi Snibgo!

The image on the drive (Jane Doe - Teaching Assistant.jpg) is the same one as shown in the screenshots.  

https://drive.google.com/drive/folders/1jkv4U7EjxlbwusGc_vEO5hbvgn-uk37R?usp=sharing

It has had certain iptc metadata fields stripped, but the relevant ones for our purposes (credit, source, specialinstructions and tranismissionreference) is present (according to exiftool and GIMP metadata view at least).

**Comment by snibgo** [comment, marked as answer]
Try it like this:
```
C:\web\im>%IMG7%magick "Jane Doe - Teaching Assistant.jpg" -format %[IPTC:2:40] info:
Teaching Assistant
```

"`magick in.jpg -verbose info:`" will tell you the numbers to use.

**Comment by Trailboy** [comment]
:)  

Ahh, after using your changes, it is now successfully using the metadata in the annotation!

Sir, thank you so much for your knowledge, yet again!

![Success01](https://user-images.githubusercontent.com/64076927/221433837-588dccd1-a2f4-4320-9cd6-aea3330f5991.JPG)

![Success02](https://user-images.githubusercontent.com/64076927/221433843-b52da2a4-e440-49d7-9a05-0ae9b0634e33.JPG)
